libraries/stdlib/js/src/kotlin/string.kt

```diff
@@ -64,3 +65,30 @@ public inline val CharSequence.size: Int get() = length
 
 @kotlin.internal.InlineOnly
 internal inline fun String.nativeReplace(pattern: RegExp, replacement: String): String = asDynamic().replace(pattern, replacement)
+
+@kotlin.internal.InlineOnly
+public actual inline fun String.compareTo(other: String, ignoreCase: Boolean): Int {
+    if (ignoreCase) {
+        val n1 = this.length
+        val n2 = other.length
+        val min = min(n1, n2) - 1
+        for (i in 0..min) {
+            var c1 = this[i]
+            var c2 = other[i]
+            if (c1 != c2) {
+                c1 = c1.toUpperCase()
+                c2 = c2.toUpperCase()
+                if (c1 != c2) {
+                    c1 = c1.toLowerCase()
+                    c2 = c2.toLowerCase()
+                    if (c1 != c2) {
+                        return c1 - c2
+                    }
+                }
+            }
+        }
+        return n1 - n2
```

Comment:
I doubt that this method of comparison is efficient, given how `Char` type and its `toUpper/LowerCase` extensions are implemented in JS. Here we make char code -> 1 char string -> upper/lower case -> back to char code conversion two times for each char. At least we can take a substring of length 1 instead of a char and avoid char code <-> string conversions.

Have you considered converting the entire strings `toUpperCase().toLowerCase()` and then comparing them? That however will probably be less efficient for long strings that differ in first several chars. 
 
Maybe we can apply some compromise approach so that the strings are split to batches each of several chars long and than each batch is transformed (toUpper -> toLower) and compared?